### PR TITLE
feat(list)!: add displayMode property to choose between flat and nested lists

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #9173

## Summary

- add `displayMode` property to choose between flat and nested lists
- update demos
- update stories
- add tests

BREAKING CHANGE: Choose a `displayMode` of "nested" or "flat" according to space and nesting requirements.

### CODE REVIEW

**Comment by driskull** [comment]
@ashetland could you give this a view? 👀

File: packages/calcite-components/src/components/list/list.tsx [hunk 1]
```diff
@@ -239,6 +241,9 @@ export class List
    */
   messages = useT9n<typeof T9nStrings>({ blocking: true });
 
+  /** Specifies the nesting behavior. */
+  @property({ reflect: true }) mode: ListMode = "flat";
```
Comment (macandcheese) [inline packages/calcite-components/src/components/list/list.tsx:245]: Should this be `displayMode` to match some other component nomenclature? This doesn't affect any selection or other functionality right, just visual?

Comment (driskull) [reply packages/calcite-components/src/components/list/list.tsx:245]: I'll defer to @jcfranco

Comment (ashetland) [reply packages/calcite-components/src/components/list/list.tsx:245]: It does affect some functionality. `nested` must be used to add the expand button and indent children. That said, I could go either way here and also defer to @jcfranco.

**Review by ashetland — APPROVED** [review]
Visuals lookin' good!

File: packages/calcite-components/src/components/list/list.e2e.ts [hunk 2]
```diff
@@ -136,153 +159,259 @@ describe("calcite-list", () => {
...
-              <calcite-list-item open label="Depth 2" description="Item 2">
-                <calcite-list drag-enabled group="my-list">
-                  <calcite-list-item label="Depth 3" description="Item 3">
-                    <calcite-list drag-enabled group="my-list"></calcite-list>
-                  </calcite-list-item>
-                  <calcite-list-item label="Depth 3" description="Item 4"></calcite-list-item>
-                </calcite-list>
-              </calcite-list-item>
-              <calcite-list-item label="Depth 2" description="Item 5"></calcite-list-item>
+              <div>
```
Comment (jcfranco) [inline packages/calcite-components/src/components/list/list.e2e.ts:221]: What's the use case for wrapped list items? If it's not a strong one, I'd suggest not supporting this to keep things simple. This also aligns with the default slot's description where we only expect items and groups. Also applies to similar `dragHandle` test.

File: packages/calcite-components/src/components/list/list.tsx (hunk 1 above)
Comment (jcfranco) [reply packages/calcite-components/src/components/list/list.tsx:245]: Good point. Let's go with `displayMode` for consistency. The only other `mode` prop I found was `input-time-zone`'s, which affects both the API and presentation.

**Review by jcfranco — APPROVED** [review]
Awesome stuff, @driskull! ✨📝✨

I have a concern regarding support for unexpected, slotted items, but aside from that, this LGTM!

File: packages/calcite-components/src/components/list/list.e2e.ts (hunk 2 above)
Comment (driskull) [reply packages/calcite-components/src/components/list/list.e2e.ts:221]: This is basically a copy from an existing test for dragHandle. I can remove the test though.

Comment (jcfranco) [reply packages/calcite-components/src/components/list/list.e2e.ts:221]: Yeah, let's remove.

**Pull request merged**